packages/maskbook/src/components/InjectedComponents/NFT/NFTBadge.tsx

```diff
@@ -50,18 +49,21 @@ const useStyles = makeStyles()({
 
 interface NFTBadgeProps extends withClasses<'root' | 'text' | 'icon'> {
     avatar: AvatarMetaDB
+    size?: number
 }
 export function NFTBadge(props: NFTBadgeProps) {
     const classes = useStylesExtends(useStyles(), props)
-    const { avatar } = props
+    const { avatar, size = 10 } = props
 
     const [amount_, setAmount_] = useState('0')
     const [symbol_, setSymbol_] = useState('')
-    useAsync(async () => {
-        const { amount, symbol } = await getNFT(avatar.address, avatar.tokenId)
+    const { value = { amount: '0', symbol: 'ETH' }, loading } = useNFT(avatar.address, avatar.tokenId)
+
+    const { amount, symbol } = value
+    useEffect(() => {
         setAmount_(amount)
         setSymbol_(symbol)
-    }, [avatar])
+    }, [amount, symbol])
```

Comment:
Break into two `useEffect`s

```ts
useEffect(() => {
        setAmount_(amount)
    }, [amount])
useEffect(() => {
        setSymbol_(symbol)
    }, [symbol])
```

Or, looks like you don't need `amount_` and `symbol_` anymore, just use `amount` and `symbol` instead direactly.